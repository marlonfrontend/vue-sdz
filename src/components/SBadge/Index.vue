<template>
  <div class="s-badge">
    <div v-if="status === 'alert' " class="--is-alert">
      <slot />
    </div>
    <div v-if="status === 'number'" class="--is-alert-number">
      <slot /><div class="--is-number">{{ number }}</div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'SBadge',

  props: {
    status: {
      type: String,
      validator: function (value) {
        return ['alert', 'number'].indexOf(value) !== -1
      },
      default: 'alert'
    },

    number: {
      type: Number,
      default: 1
    }
  }
}
</script>

<style lang="scss">
@import "./src/styles/_index.scss";

.s-badge {
  color: color(base, light);
  position: relative;

  & > .--is-alert{
    &::after{
      content: '';
      position: absolute;
      right: 0px;
      top: -2px;
      height: 8px;
      width: 8px;
      background: color(highlight, base);
      border-radius: 50px;
    }
  }

  & > .--is-alert-number > .--is-number{
    position: absolute;
    right: -12px;
    top: -11px;
    padding: 3px 8px;
    background: color(highlight, base);
    border-radius: 50px;

    font-size: 12px;
    line-height: 15px;
    font-weight: 500;
  }
}
</style>
