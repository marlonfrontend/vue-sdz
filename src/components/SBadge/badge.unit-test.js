import { mount } from '@vue/test-utils'

import SBadge from './Index.vue'

/* eslint-disable */
describe('SBadge', () => {

  let cmp;

  beforeEach(() => {
    cmp = mount(SBadge, {
      propsData: {
        dot: true
      }
    });
  });


  test('is instantiated', () => {
    expect(cmp.exists()).toBeTruthy()
  })
  

  test("has the expected html structure", () => {
    expect(cmp.element).toMatchSnapshot();
  })

  test('check content', () => {
    const wrapper = mount(SBadge, { propsData: { content: '5' } })

    const content = wrapper.find('div > div')
    expect(content.exists()).toBe(true)
    expect(content.text()).toBe('5')
  })

  test('check dot class', () => {
    
    const div = cmp.find('div > div')
    expect(div.classes('--dot')).toBe(true)
    expect(div.classes()).toContain('--dot')
  })

})
