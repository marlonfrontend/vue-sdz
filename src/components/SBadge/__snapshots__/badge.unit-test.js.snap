// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SBadge has the expected html structure 1`] = `
<div
  class="s-badge"
>
  <div
    class="content --dot --primary"
  >
    
  </div>
   
</div>
`;
